Libraries/ActionSheetIOS/ActionSheetIOS.js

```diff
@@ -25,6 +36,13 @@ var ActionSheetIOS = {
       typeof callback === 'function',
       'Must provide a valid callback'
     );
+
+    var processedTintColor = this.processTintColor(options.tintColor);
+
+    if (processedTintColor) {
+      options.tintColor = processedTintColor;
+    }
```

Comment:
He's right that you shouldn't call processColor() multiple times, but I also don't think there's any benefit to checking its result before assigning it.

In the code as you've written it, if `options.tintColor` is some invalid string that causes processedTintColor to return null or zero, you're actually leaving `options.tintColor` as the original string instead of overwriting it with transparent, which will then crash the native code that's expecting a number instead of a string.

It's perfectly safe to just write

    options.tintColor = processColor(options.tintColor);

That should ensure that whatever garbage value options.tintColor goes in as, it will come out as a sanitised value. If for some reason it doesn't, that should be fixed in processColor.js, not in ActionSheetIOS.js.